lib/local_resource.rb

```diff
@@ -25,7 +25,7 @@ def encoding
   end
 
   def find_file_type
-    @file_type ? @file_type.split('/').last : Pathname.new(uri.path).extname
```

Comment:
There are places where we pass encoding as the file type. hence this is intended.  
cc: @pranavrajs